matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/MXOlmDevice.kt

```diff
@@ -133,7 +143,18 @@ internal class MXOlmDevice @Inject constructor(
      * @return The maximum number of one-time keys the olm account can store.
      */
     fun getMaxNumberOfOneTimeKeys(): Long {
-        return store.getOlmAccount().maxOneTimeKeys()
+        return doWithOlmAccount { it.maxOneTimeKeys() }
+    }
+
+    /**
+     * Olm account access should be synchronized
+     */
+    private fun <T> doWithOlmAccount(block: (OlmAccount) -> T): T {
+        return store.getOlmAccount().let { olmAccount ->
```

Comment:
To avoid future mistake I would move the fun `doWithOlmAccount` to the store and remove getOlmAccount(), WDYT?

Reply:
yeah good idea